torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -264,109 +280,101 @@ def ReplaceMoreTtManually(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
                     new_kwargs = {
                         "fill_value": args[1],
                         "device": TtnnDevice(),
                     }
                     full = g.call_function(
                         ttnn.full,
-                        args=(tuple(node_metadata.size()),),
+                        args=(tuple(shape),),
                         kwargs=new_kwargs,
                     )
-                    # Intermediate node meta is not preserved, this ensures retention
-                    full.meta["val"] = node.meta["val"]
                     to_layout = g.call_function(
                         ttnn.to_layout, (full,), {"layout": TtnnTileLayout()}
                     )
-                    recip = g.call_function(ttnn.reciprocal, (to_layout,), {})
-                else:
-                    recip = g.call_function(ttnn.reciprocal, (args[1],), {})
-                # Intermediate node meta is not preserved, this ensures retention
-                recip.meta["val"] = node.meta["val"]
-                new_node = g.call_function(ttnn.mul, (args[0], recip), {})
-                new_node.meta["val"] = node.meta["val"]
-                node.replace_all_uses_with(
-                    new_node,
-                    delete_user_cb=lambda node: node != new_node,
-                )
+                    new_node = g.call_function(
+                        ttnn.sub, args=(to_layout, args[0]), kwargs={}
+                    )
+                    new_nodes.append(new_node)
+            if node.target == torch.ops.aten.div.Tensor:
+                # ttnn.recip does not support scalars. Call an ttnn.full and pass that to ttnn.recip
+                # TODO(kevinwuTT): Use a ttnn equivalent
+                node_metadata = node.meta["val"]
+                shape = node_metadata.size()
+                # If last dim == 1, then the follow error will appear:
+                # Page size must be divisible by sizeof(uint32_t) because buffers hold uint32_t values
+                if shape[-1] != 1 and has_valid_page_size(shape):
+                    if isinstance(args[1], float):
+                        new_kwargs = {
+                            "fill_value": args[1],
+                            "device": TtnnDevice(),
+                        }
+                        full = g.call_function(
+                            ttnn.full,
+                            args=(tuple(shape),),
+                            kwargs=new_kwargs,
+                        )
+                        to_layout = g.call_function(
+                            ttnn.to_layout, (full,), {"layout": TtnnTileLayout()}
+                        )
+                        recip = g.call_function(ttnn.reciprocal, (to_layout,), {})
+                    else:
+                        recip = g.call_function(ttnn.reciprocal, (args[1],), {})
+                    new_node = g.call_function(ttnn.mul, (args[0], recip), {})
+                    new_nodes.append(new_node)
```

Comment:
Not a matter of this PR. But want to leave thoughts here.

I think of multiple types of conversions:
1. **Trivial conversion** is a conversion which allows full 1-1 mapping. In this case it is just a record in a map
2. **Arg Order conversion** is a conversion which maps close to 1-1 but requires arguments reordering
3. **Args Change conversion** is a conversion which maps close to 1-1 but requires arguments to be modified
4. **Lowering conversion** is a conversion which maps 1 aten into multiple TT-NN ops

I don't like seeing a long file for 400 lines of code with all conversions at the same place.
Ideally conversions are separate, stored in some registry, and generic conversion mechanism just pulls the required conversion.